integration-tests/tests/websocket/websockets.spec.js

```diff
@@ -14,40 +14,122 @@ See the License for the specific language governing permissions and
-    await browser.assert.strictEqual(wsConnectionSpan.tags['ot.status_code'], 'UNSET'); //pointless check, always OK
+    await browser.assert.strictEqual(wsConnectionSpan.tags['ot.status_code'], 'UNSET');
     await browser.assert.strictEqual(wsConnectionSpan.tags['error'], undefined);
   },
   'websocket url can be ignored': async function(browser) {
-    const UNSUPPORTED_BROWSERS = ['Safari'];
-    const currentBrowser = browser.options.desiredCapabilities.browserName;
-    if (UNSUPPORTED_BROWSERS.includes(currentBrowser)) {
+    if (isSafari(browser)) {
       return;
     }
-    
-    browser.globals.clearReceivedSpans();
     await browser.url(browser.globals.getUrl('/websocket/websocket-ignored.ejs'));
-    await browser.click('#connectWebsocketsBtn');
+    await browser.click('#connectWs');
 
     await browser.globals.findSpan(span => span.name === 'websocket-guard-span');   
     await browser.assert.not.ok(browser.globals.receivedSpans.find(span => span.name === 'connect'));
-  }
+  },
+  'sending send and on message create a span': async function(browser) {
+    if (isSafari(browser)) {
+      return;
+    }
+    await browser.url(browser.globals.getUrl('/websocket/websocket.ejs'));
+    await browser.click('#connectWs');
+    // check for connect span so connection has time to initialize otherwise send will fail
+    const wsConnectionSpan = await browser.globals.findSpan(span => span.name === 'connect');     
+    await browser.assert.ok(!!wsConnectionSpan, 'Connect span missing');     
+    await browser.click('#sendWs');
+    const wsMessage = await browser.globals.findSpan(span => span.name === 'onmessage');
+    await browser.assert.ok(!!wsMessage);
+    await browser.assert.strictEqual(wsMessage.kind, 'CONSUMER');
+    await browser.assert.strictEqual(wsMessage.tags['component'], 'websocket');
+    await browser.assert.strictEqual(wsMessage.tags['protocol'], '');
+    await browser.assert.strictEqual(wsMessage.tags['http.url'], browser.globals.wsBase);
+    await browser.assert.strictEqual(wsMessage.tags['http.response_content_length'], '7');
+
+    const wsSend = browser.globals.receivedSpans.find(span => span.name === 'send');
+    await browser.assert.ok(!!wsSend);
+    await browser.assert.strictEqual(wsSend.kind, 'PRODUCER');
+    await browser.assert.strictEqual(wsSend.tags['component'], 'websocket');
+    await browser.assert.strictEqual(wsSend.tags['protocol'], '');
+    await browser.assert.strictEqual(wsSend.tags['http.url'], browser.globals.wsBase);
+    await browser.assert.strictEqual(wsSend.tags['http.request_content_length'], '12');
+  },
+  'websocket constructor errors are captured': async function(browser) {
+    if (isSafari(browser)) {
+      return;
+    }
+    await browser.url(browser.globals.getUrl('/websocket/websocket-construct-errors.ejs'));
+    await browser.click('#connectWs');
+    const wsConnectionSpan = await browser.globals.findSpan(span => span.name === 'connect', -5000);     
```

Comment:
Why do we need additional 5s for this span to arrive?

Reply:
Making of ws connected was slow sometimes and caused a flaky test iirc